server/cluster/cluster.go

```diff
@@ -521,12 +531,20 @@ func (c *RaftCluster) fetchStoreConfigFromTiKV(statusAddress string) (*config.St
 	} else {
 		url = fmt.Sprintf("%s://%s/config", "http", statusAddress)
 	}
-	resp, err := c.httpClient.Get(url)
+	ctx, cancel := context.WithTimeout(ctx, clientTimeout)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, bytes.NewBuffer(nil))
+	if err != nil {
+		cancel()
+		return nil, fmt.Errorf("failed to create store config http request: %w", err)
```

Comment:
Do we need to update the store instead of always trying them?

Reply:
no, it will remove this store if the function return err in line 475.

Reply:
Will line 475 delete all stores?

Reply:
only delete the bad store.

Reply:
But if the PD has a network partition with all TiKV within the sync period, once it recovers, will it have a problem?

Reply:
I think no, pd will refresh all store state if all the previous stores are bad.

Reply:
make sense